raiden/utils/__init__.py

```diff
@@ -288,3 +288,20 @@ def safe_gas_limit(*estimates) -> int:
     """
     calculated_limit = max(gas or 0 for gas in estimates)
     return int(calculated_limit * constants.GAS_FACTOR)
+
+
+def spawn_and_link_with_parent(func, *args, **kwargs):
+    """ Spawn a new greenlet and link it to the current thread instead of the hub. """
+    parent = gevent.getcurrent()
+
+    # closure for the parent greenlet
+    def on_error(subtask):
+        assert parent is not None
+        # in case the parent task already finished, we escalate to the hub
+        if parent.ready():
+            gevent.get_hub().throw(subtask.exception)
```

Comment:
I don't think this is necessary at all. gevent has it's [own spawn implementation](https://github.com/gevent/gevent/blob/6d8d6fb19943ff53c75f153ee22310eb165beb9f/src/gevent/__init__.py#L93), which automatically links the [new greenlet to the hub](https://github.com/gevent/gevent/blob/6d8d6fb19943ff53c75f153ee22310eb165beb9f/src/gevent/greenlet.py#L213), and greenlets propagate the exceptions in the parent tree [ref](https://greenlet.readthedocs.io/en/latest/#parents):

> The parent greenlet is initially the one in which the greenlet was created (this can be changed at any time). The parent is where execution continues when a greenlet dies. This way, greenlets are organized in a tree. Top-level code that doesn’t run in a user-created greenlet runs in the implicit “main” greenlet, which is the root of the tree.

Meaning that if `parent` is already dead, then the exception will be raised to the hub anyways. Additionally, what does this accomplish? The hub swallows exceptions which is exactly what we don't want.